### DISCUSSION TITLE
Start opens on left side even when centered

### DISCUSSION THREAD

**Discussion by currall** [valinet/ExplorerPatcher#1036] [Q&A]
Clicking in the left bottom corner opens start menu even when taskbar and start menu are both set to centered. Happens even when weather isnt there.

https://user-images.githubusercontent.com/29579820/157472523-233dcd00-28e9-4624-b60b-fffd270f32f6.mp4

**Comment by valinet** [comment]
That's actually intended functionality. Aka it's not a bug, it's by design.

The logic is that, when you center the taskbar, that precious corner is dead (even if the weather is there, the corner is still technically dead by default because the container of the weather widget has a thin left border of 1 px which does nothing).

This is despite muscle memory telling everyone Start should open when you press the lower left corner. So, I made it open Start.

The question left was where to open Start when invoked in this scenario: since you moved the mouse to the left side anyway, it made sense to have it open there, so you don't have to navigate all the way back to the center of the screen. And so, when invoked in this way in this particular scenario, Start will override whatever setting you have and always open at the left. I took actual time to code this out, so, as I said, it's intended functionality.

Now, if it happens with any other interaction, then it's a bug. Start should open at the setting you chose when you click the Start button or press the `Win` key or `Ctrl`+`Esc`.

This exception here I made since I don't take anything, but rather give. The corner would otherwise be dead anyway, since, with the "Centered with Start button" mode for centering the taskbar, the Start button is near the icons at the center, so nothing's there really.

Is there a compelling reason to change this behavior? That above is the logic behind it.

Thanks

**Reply by currall** [reply]
I suppose that makes sense, but I expected it to either open weather or do nothing and people who want the centered start button probably arent bothered about muscle memory since they would put the start button in the left if they did.

I guess its a good option to have but it would be nice if it was optional

**Reply by valinet** [reply]
Okay, it will be an option disabled by default in the next update. Sorta wanted to do it like this since the beginning, I thought it would make sense for more people. Oh well...

**Reply by Poopooracoocoo** [reply]
Could just be a vocal minority for all we know.

**Reply by valinet** [reply]
Indeed. The most important thing is not to turn this forum into a wall of reports similar to this one. A lot of people send me messages already for the fact that the taskbar with labels is the default, which they did not expect... Anyway, I have made the thing optional for now. Seekers will benefit from it, basically. To enable it, set `UndeadStartCorner` in `HKCU\SOFTWARE\ExplorerPatcher`:

* `1` will have Start open to whatever setting you have configured for it in Properties - Start menu when clicking the corner
* `2` will have Start open to the left when clicking the corner, regardless of the setting configured in Properties - Start menu

This should be fine for now, I keep it on `2` on my system and still am able to keep in sync with the upstream ;)
